app/domain/authentication/resource_restrictions/fetch_resource_annotations.rb

```diff
@@ -0,0 +1,46 @@
+require 'command_class'
+
+module Authentication
+  module ResourceRestrictions
+
+    FetchResourceAnnotations = CommandClass.new(
+      dependencies: {
+        role_class: ::Role,
+        resource_class: ::Resource
+      },
+      inputs: %i[account role_name ignore_empty_annotations]
+    ) do
+
+      def call
+        fetch_resource_annotations
+      end
+
+      private
+
+      def fetch_resource_annotations
+        resource_annotations
+      end
+
+      def resource_annotations
+        resource.annotations.each_with_object({}) do |annotation, result|
+          annotation_values = annotation.values
+          value = annotation_values[:value]
+          next if value.blank? and @ignore_empty_annotations
```

Comment:
i'm ok with this approach but why not raise an error? 
how about this:
```
        if value.blank?
          if @ignore_empty_annotations
            next
          else          
            raise Errors::Authentication::ResourceRestrictions::EmptyAnnotationGiven, restriction.name 
       end

```

Reply:
this way you don't need to handle it for jwt and we can utilize this in other places by just passing `ignore_empty_annotations: false`

Reply:
Because in JWT we want to throw exception for annotation that is empty but relevant (according to its name) to the vendor (service id) trying to authenticate. The vendor check happens only after the extraction. So in JWT we will want to pass on the empty annotation in order to handle it later.

Reply:
ok thanks for clarifying

Reply:
For the case of supporting third behaviour of throwing exception i think it better to inject another class doing it instead of featch_resource_annotations.

But i can also agree if you say for a const specifiying it that have three states (one for ignore, one for error, and one for passing it empty as is). In jwt i need the passing it as is to check later if exception need to be thrown (in the validator)